### DISCUSSION TITLE
Problems with 'Manage Access' for External DB Tables

### DISCUSSION THREAD

**Discussion by rhrish** [Budibase/budibase#3028] [Help] [answered]
I have created a form that is public facing and this form has a countries drop down that gets populated from a mariaDB table. 

I was trying to set this table and a couple of other tables to Public - Read access so that it functions properly. However I am stuck with this screen when I click on 'Manage Access' for this table and many other tables as well.

![Screen Shot 2021-10-17 at 6 58 04 AM](https://user-images.githubusercontent.com/119729/137606755-75d39846-48f4-4b83-b0d3-50f22b0db7a8.png)

Is there a workaround by which I can add READ only for public to this table? Like export the application, edit the exported file directly to add access attributes to the table and imported it back? What should I look for in the exported file? cc: @mjashanks

**Comment by rhrish** [comment]
On further debugging, I discovered that this API endpoint /api/permission/datasource_plus_... is giving the following response {"message":"Cannot read property 'indexOf' of undefined","status":500}

The 'Manage Access' worked for one table (the first table), but all subsequent tables have the above response.

**Comment by rhrish** [comment, marked as answer]
Update: For anyone else stuck with the same problem.

Here's a work around. 

- Export your app
- Find the role_PUBLIC block in the exported file. It should look something like this.

```
{
  "_id": "role_PUBLIC",
  "_rev": "1-0abcdeflkjd92jksdh1234567893",
  "name": "Public",
  "permissionId": "public",
  "permissions": {
    "datasource_plus_cabcdf0eabcdefgh8711234567897dcd__Countries": [
      "read"
    ]
  }
}
```

- Add your table's name to the permissions object like so

```
  "permissions": {
    "datasource_plus_cabcdf0eabcdefgh8711234567897dcd__Countries": [
      "read"
    ],
   "datasource_plus_abcdefgheabcdefgh8711234567897dcd__MyOtherTable": [
      "read"
    ],
  }
```

- Save the file and import it as a new application into budibase

**Reply by mjashanks** [reply]
Thanks so much for reporting back on the workaround. I'll try to reproduce the issue and get it logged :)

**Reply by mjashanks** [reply]
Logged this here: https://github.com/Budibase/budibase/issues/3093

Thanks for reporting :)

**Reply by mjashanks** [reply]
And... fixed! Available in next release.

**Reply by rhrish** [reply]
Excellent! Thanks again!!

**Comment by rhrish** [comment]
The public facing form I use also makes use of a couple of custom queries. Is there a way I can make the access of a query datasource public-read? I tried adding them to the permissions block but it doesn't seem to work.

**Reply by mjashanks** [reply]
Unfortunately not at the moment

**Reply by rhrish** [reply]
Sigh! I hope it gets added in the future!
